Cors with additional headers

MichalCz opened the pull request:
I added a small mod that allows to pass additional allowed request headers to corser.

This is done by passing: --corsheaders="Header,List,Separated,By,Commas".

indexzero commented on bin/http-server:
```diff
@@ -80,6 +81,10 @@ function listen(port) {
   }
 
+  if (argv.corsheaders) {
+    options.cors_headers = argv.corsheaders;
```
`options.corsHeaders`.

indexzero commented on bin/http-server:
```diff
@@ -20,6 +20,7 @@ if (argv.h || argv.help) {
     "  -e --ext           Default file extension if none supplied [none]",
     "  -s --silent        Suppress log messages from output",
     "  --cors             Enable CORS via the 'Access-Control-Allow-Origin' header",
+    "  --corsheaders      Additional CORS headers",
```
This doesn't need to be a separate option. We should simply expect `--cors` to also be a string sometimes.

indexzero commented:
@MichalCz will accept this once:

1. Tests are passing (mostly style it seems)
2. There are tests for this feature
3. It is implemented as a single option (e.g. `--cors` or `--cors "Header,List,Separated,By,Commas"`). No need for `--corsheaders`.

MichalCz commented:
Will fix next week. :)

On Tue, 22 Sep 2015 03:02 Charlie Robbins <notifications@github.com> wrote:

> @MichalCz <https://github.com/MichalCz> will accept this once:
>
>    1. Tests are passing (mostly style it seems)
>    2. There are tests for this feature
>    3. It is implemented as a single option (e.g. --cors or --cors
>    "Header,List,Separated,By,Commas"). No need for --corsheaders.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/indexzero/http-server/pull/150#issuecomment-142150656>
> .
>

MichalCz commented:
Hey @indexzero 

I rebased the changes, added tests, fixed changes. If you had any other requests let me know.

BTW - I still use a sepatate option in the javascript interface so that there's no need to check type in the http-server.js code, I changed command line usage only.

/Mike

MichalCz commented:
Oh, wait - one more fix to do...

/Mike

MichalCz commented:
@indexzero - code should now comply with what you wanted. :)

/Mike

indexzero commented:
Thanks!

danielweck commented:
Nice one!
Fixes https://github.com/indexzero/http-server/issues/154
